pkg/registry/service/rest.go

```diff
@@ -265,6 +275,97 @@ func (*REST) NewList() runtime.Object {
 	return &api.ServiceList{}
 }
 
+func (rs *REST) healthCheckNodePortUpdate(oldService, service *api.Service) (bool, error) {
+	// Health Check Node Port handling during updates
+	//
+	// Case 1. Transition from globalTraffic to OnlyLocal for the ESIPP annotation
+	//
+	//   Allocate a health check node port or attempt to reserve the user-specified one, if provided.
+	//   Insert health check node port as an annotation into the service's annotations
+	//
+	// Case 2. Transition from OnlyLocal to Global for the ESIPP annotation
+	//
+	//   Free the existing healthCheckNodePort and clear the health check nodePort annotation
+	//
+	// Case 3. No change (Global ---stays--> Global) but prevent invalid annotation manipulations
+	//
+	//   Reject insertion of the "service.alpha.kubernetes.io/healthcheck-nodeport" annotation
+	//
+	// Case 4. No change (OnlyLocal ---stays--> OnlyLocal) but prevent invalid annotation manipulations
+	//
+	//   Reject deletion of the "service.alpha.kubernetes.io/healthcheck-nodeport" annotation
+	//   Reject changing the value of the healthCheckNodePort annotation
+	//
+	oldServiceHasHealthCheckNodePort := apiservice.NeedsHealthCheck(oldService)
+	oldHealthCheckNodePort := apiservice.GetServiceHealthCheckNodePort(oldService)
+
+	assignHealthCheckNodePort := shouldCheckOrAssignHealthCheckNodePort(service)
+	requestedHealthCheckNodePort := apiservice.GetServiceHealthCheckNodePort(service)
+
+	switch {
+	case !oldServiceHasHealthCheckNodePort && assignHealthCheckNodePort:
+		glog.Infof("Transition from Global LB service to OnlyLocal service")
+		if requestedHealthCheckNodePort > 0 {
+			// If the request has a health check nodePort in mind, attempt to reserve it
+			err := rs.serviceNodePorts.Allocate(int(requestedHealthCheckNodePort))
+			if err != nil {
+				errmsg := fmt.Sprintf("Failed to allocate requested HealthCheck nodePort %v:%v",
+					requestedHealthCheckNodePort, err)
+				el := field.ErrorList{field.Invalid(field.NewPath("ObjectMeta", "Annotations"),
```

Comment:
Maybe github is broken but i don't see the change - should be `field.NewPath("metadata", "annotations")` in all these checks